components/Ink/src/MDCInkView.m

```diff
@@ -103,30 +122,69 @@ - (void)setCustomInkCenter:(CGPoint)customInkCenter {
   self.inkLayer.customInkCenter = customInkCenter;
 }
 
-- (MDCInkLayer *)inkLayer {
-  return (MDCInkLayer *)self.layer;
+- (MDCLegacyInkLayer *)inkLayer {
+  return (MDCLegacyInkLayer *)self.layer;
 }
 
 - (void)startTouchBeganAnimationAtPoint:(CGPoint)point
                              completion:(MDCInkCompletionBlock)completionBlock {
-  [self.inkLayer spreadFromPoint:point completion:completionBlock];
+  if (self.usesLegacyInkRipple) {
+    [self.inkLayer spreadFromPoint:point completion:completionBlock];
+  } else {
+    self.startInkRippleCompletionBlock = completionBlock;
+    MDCInkLayer *inkLayer = [MDCInkLayer layer];
+    inkLayer.inkColor = self.inkColor;
+    inkLayer.maxRippleRadius = self.maxRippleRadius;
+    inkLayer.animationDelegate = self;
+
+    switch (self.inkStyle) {
+      case MDCInkStyleBounded:
+        self.clipsToBounds = YES;
+        break;
+      case MDCInkStyleUnbounded:
+        self.clipsToBounds = NO;
+        break;
+    }
+
+    inkLayer.opacity = 0;
+    inkLayer.frame = self.bounds;
+    [self.layer addSublayer:inkLayer];
+    [inkLayer startAnimationAtPoint:point];
+    self.activeInkLayer = inkLayer;
+  }
 }
 
-- (void)startTouchEndedAnimationAtPoint:(__unused CGPoint)point
+- (void)startTouchEndedAnimationAtPoint:(CGPoint)point
                              completion:(MDCInkCompletionBlock)completionBlock {
-  [self.inkLayer evaporateWithCompletion:completionBlock];
+  if (self.usesLegacyInkRipple) {
+    [self.inkLayer evaporateWithCompletion:completionBlock];
+  } else {
+    self.endInkRippleCompletionBlock = completionBlock;
+    [self.activeInkLayer endAnimationAtPoint:point];
+  }
 }
 
 - (void)cancelAllAnimationsAnimated:(BOOL)animated {
-  [self.inkLayer resetAllInk:animated];
+  if (self.usesLegacyInkRipple) {
+    [self.inkLayer resetAllInk:animated];
+  } else {
+    if (animated) {
+      for (CALayer *layer in self.layer.sublayers) {
+        if ([layer isKindOfClass:[MDCInkLayer class]]) {
+          MDCInkLayer *inkLayer = (MDCInkLayer *)layer;
+          [inkLayer endAnimationAtPoint:CGPointZero];
+        }
+      }
+    } else {
+      self.layer.sublayers = nil;
```

Comment:
This isn't necessarily equivalent to the above code - if someone added a custom layer to this layer then this line would nuke it. It might be safer if we run the for loop in both animated/non-animated cases and only remove the sublayers if we're sure they are ink layers.

Reply:
Done.